src/utils/pluginDriver.ts

```diff
@@ -192,3 +226,95 @@ export function createPluginDriver(
 
 	return pluginDriver;
 }
+
+export function createPluginCache(cache: SerialisablePluginCache): PluginCache {
+	return {
+		has(id: string) {
+			const item = cache[id];
+			if (!item) return false;
+			item[0] = 0;
+			return true;
+		},
+		get(id: string) {
+			const item = cache[id];
+			if (!item) return undefined;
+			item[0] = 0;
+			return item[1];
+		},
+		set(id: string, value: any) {
+			cache[id] = [0, value];
+		},
+		delete(id: string) {
+			return delete cache[id];
```

Comment:
Objects with lots of key changes are treated as "dictionary objects" with hash table keys (which is exactly what we need here). In that case the delete cost is the delete cost of the hash table which shouldn't be a problem. Also we do want to evict unused keys to avoid cache bloat.